### PR TITLE
Refactor caching

### PR DESCRIPTION
**Pull request by dhiemstra**
* Always use `MemoryCache` cache and use `NullCache` to disable caching completely
* Add some tests

@aaronmallen hope this doesn't break your app :)

### CODE REVIEW

File: lib/esi.rb [hunk 1]
```diff
@@ -70,12 +73,12 @@ module Esi
     timeout: 60,
     client_id: nil,
     client_secret: nil,
-    cache: nil,
+    cache: ActiveSupport::Cache::MemoryStore.new,
```
Comment (aaronmallen) [inline lib/esi.rb]: I don't know how I feel about this... this makes caching opt-out instead of opt-in

Comment (aaronmallen) [reply lib/esi.rb]: perhaps default to `:null_store` instead?

Comment (aaronmallen) [reply lib/esi.rb]: Also lets use `ActiveSupport::Cache.lookup_store` instead.

**Comment by dhiemstra** [comment]
@aaronmallen let me make another PR to address your feedback this weekend. Just merging this so i can create another PR with a lot of fixes.

**Pull request merged**